File: old-ocaml/bootloaders/bootloader.ml
(* arch-tag: Bootload setup
* Copyright (c) 2004 John Goerzen
*)

open Unix;;
open Shellutil;;
open Dfsutils;;
open Archsupport;;

(** Returns a tuple of (mkisofsargs, post-mkisofsfunc) *)
let install cp workdir target =
  match get cp "bootloader" with
  "grub-no-emul" -> Grub.grub_eltorito cp target
  | "grub-hd" -> Grub.grub_hd cp workdir target
  | "aboot" -> Aboot.aboot cp target
  | "yaboot" -> Yaboot.yaboot cp workdir target
  | _ -> ( p("Invalid bootloader specified"); exit 2; ([], fun _ _ _ _ -> ()))
;;

File: Bootloader.hs
{- dfsbuild: CD image builder
Copyright (c) 2006 John Goerzen
Please see COPYRIGHT for more details
-}

module Bootloader where
import qualified Bootloader.Grub
import Utils
--import qualified Bootloader.Aboot
--import qualified Bootloader.Yaboot

install :: DFSEnv -> IO ([String], DFSEnv -> IO ())
install env =
    case eget env "bootloader" of
      "grub-no-emul" -> Bootloader.Grub.grub_eltorito env
      "grub-hd" -> Bootloader.Grub.grub_hd env
--      "aboot" -> Bootloader.Aboot.aboot env
--      "yaboot" -> Bootloader.Yaboot.yaboot env
      x -> fail $ "Invalid bootloader: " ++ x
